waku.md

```diff
@@ -459,9 +459,13 @@ Notes useful for implementing Waku mode.
 
 Summary of main differences between this spec and Whisper v6, as described in [EIP-627](https://eips.ethereum.org/EIPS/eip-627):
 
-- RLPx subprotocol is changed from `shh/6` to `waku/0`
-- Light node capability
-- Whisper Mail Server and Whisper Mail Client implemented
+- RLPx subprotocol is changed from `shh/6` to `waku/0`.
+- Light node capability is added.
+- Optional rate limiting is added.
+- Status packet has following additional parameters: light-node,
+confirmations-enabled and rate-limits
+- Mail Server and Mail Client functionality is now part of the specification.
+- P2P Message packet contains a list of envelopes instead of a single envelope.
```

Comment:
"List of envelopes"

OR

"List of envelopes OR single envelope"

Reply:
Yeah, I don't really like that part. I was going to suggest to change the spec now that it is in Waku, but I realized that it can actually still be interpreted as not something from the Waku spec, but something you might do to just remain compatible with EIP-627. 

`In order to stay compatible with EIP-627, a peer receiving historic message MUST handle both cases.`

So to be unambiguous, I would just suggest to get rid of that whole EIP-627 remain compatible part and allow only the list in Waku. What do you think?

Reply:
I'd be fine with that, as long as we don't break existing network and things work with Waku later.

@adambabik wdyt

Reply:
Currently, status-go uses a list of envelopes everywhere but for backward compatible with EIP-627 it also supports single envelope. We can drop a single envelope I believe if we are ok to break compatibility with EIP-627 which I guess is a case.

Reply:
Considering waku/0 is supposed to be EIP-627 + ad hoc stuff we added, I wonder if it makes sense to leave that packet code there?

Then we can *append* and have another packet code that works more clean, instead of having to think about edge cases like this.

IDK, what do you guys think?

Reply:
Or maybe this is yakshaving... just thought I'd raise it, happy to go either way.

Reply:
I'd be OK with the creation of a new packet for this. But I'll merge this for now, and if we decide to do so we can update.